x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStore.java

```diff
@@ -327,7 +327,7 @@ private void buildThenMaybeCacheRole(
             restrictedIndices,
             listener.delegateFailureAndWrap((delegate, role) -> {
                 if (role != null && tryCache) {
-                    try (ReleasableLock ignored = roleCacheHelper.acquireUpdateLock()) {
+                    try (ReleasableLock ignored = roleCacheHelper.acquireReadLock()) {
```

Comment:
this is just a name change, but we should probably see (in a different PR) if this should be a write lock to prevent against deadlock.